crates/oxc_mangler/src/lib.rs

```diff
@@ -330,12 +331,17 @@ impl<'t> Mangler<'t> {
             let mut slot = slot_liveness.len();
 
             reusable_slots.clear();
+            // Pre-allocate space for reusable slots to avoid multiple allocations
+            reusable_slots.reserve(tmp_bindings.len());
+            
+            // Collect reusable slots more efficiently by avoiding repeated contains() calls
+            let scope_index = scope_id.index();
             reusable_slots.extend(
                 // Slots that are already assigned to other symbols, but does not live in the current scope.
                 slot_liveness
                     .iter()
                     .enumerate()
-                    .filter(|(_, slot_liveness)| !slot_liveness.contains(scope_id.index()))
+                    .filter(|(_, slot_liveness)| !slot_liveness.contains(scope_index))
```

Comment:
Precalculating `scope_index` here is good. Compiler may well be smart enough to do that itself, but it's better to be explicit and not rely on that.